Use IMDSv2 when retrieving the instance ID

jsf9k opened the pull request:
## 🗣 Description ##

This pull request modifies the `00_setup_freeipa.sh` script to use IMDSv2 when retrieving the instance ID from the AWS meta-data service.

## 💭 Motivation and context ##

We want to enforce IMDSv2-only on our instances to satisfy cisagov/cool-system#174 and cisagov/cool-system#175, and this change will help us with that.

## 🧪 Testing ##

* All `pre-commit` hooks and `molecule` tests pass
* I verified that the new code will work on a running instance in our COOL staging environment.

## ✅ Checklist ##

* [x] This PR has an informative and human-readable title.
* [x] Changes are limited to a single goal - _eschew scope creep!_
* [x] All relevant type-of-change labels have been added.
* [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
* [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
* [x] All new and existing tests pass.

jsf9k committed:
Use IMDSv2 when retrieving the instance ID

We want to enforce IMDSv2-only on our instances to satisfy
cisagov/cool-system#174 and cisagov/cool-system#175, and this change
will help us with that.

dav3r commented on files/00_setup_freeipa.sh:
```diff
@@ -151,11 +151,18 @@ function setup {
+    http://169.254.169.254/latest/api/token)
+  instance_id=$(curl --silent \
+      --header "X-aws-ec2-metadata-token: $imds_token" \
+    http://169.254.169.254/latest/meta-data/instance-id)
```
Is it worthwhile to add a check to ensure that the `instance_id` is valid (or at least not empty) before we attempt to use it?

jsf9k replied:
What should we do if it doesn't match an appropriate regex, say?

dav3r replied:
If we have confidence in our regex, then I say we print a useful error message and bail out.

jsf9k replied:
I don't think it makes sense to bail out, since by that point we have already set up the FreeIPA server and have completed 95% of the script, but I can print an error message stating that the attempt to create the alternate hostname in FreeIPA has failed.

dav3r replied:
Agreed- that makes more sense.  I thought (without checking) that this code was at the beginning of the script, not so close to the end.

jsf9k replied:
See commits edd4679 and cisagov/ansible-role-freeipa-client@1a3b3fb.

jsf9k committed:
Add some code to verify that the instance ID is valid

If it is not then we print an error message and do not attempt to
create the principal alias containing the instance ID.

Co-authored-by: Dave Redmin <david.redmin@beta.dhs.gov>

dav3r reviewed (approved):
Very nice! 🪕

mcdonnnj reviewed (approved):
Makes sense. ✔
